tests/external_approval_test.go

```diff
@@ -0,0 +1,168 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/bytebase/bytebase/tests/fake"
+)
+
+func TestExternalApprovalFeishu(t *testing.T) {
+	t.Parallel()
+	a := require.New(t)
+	ctx := context.Background()
+	ctl := &controller{}
+	dataDir := t.TempDir()
+	err := ctl.StartServer(ctx, &config{
+		dataDir:                 dataDir,
+		vcsProviderCreator:      fake.NewGitLab,
+		feishuProverdierCreator: fake.NewFeishu,
+	})
+	a.NoError(err)
+	defer ctl.Close(ctx)
+	err = ctl.Login()
+	a.NoError(err)
+
+	err = ctl.setLicense()
+	a.NoError(err)
+
+	// close existing issues
+	issues, err := ctl.getIssues(api.IssueFind{})
+	a.NoError(err)
+	for _, issue := range issues {
+		patchedIssue, err := ctl.patchIssueStatus(api.IssueStatusPatch{
+			ID:     issue.ID,
+			Status: api.IssueCanceled,
+		})
+		a.NoError(err)
+		a.Equal(api.IssueCanceled, patchedIssue.Status)
+	}
+
```

Comment:
Is this possible to set the url override in the server profile and test server gets its own localhost path?